src/task.py

```diff
@@ -237,17 +237,110 @@ def get_ls_country_biome_kbas(kba_id):
 
+                def calc_hab_areas_state():
+                    country_code = ee.Number(country.get("isonumeric")).int()
+
+                    area = (
+                        ee.Image.pixelArea().divide(1000000).updateMask(self.watermask)
+                    )
+                    str_hab_area = area.updateMask(self.structural_habitat).rename(
+                        "str_hab_area"
+                    )
+                    habitat_image = (
+                        self.scl_image.addBands(
+                            [
+                                self.occupied_habitat_image.rename("occupied_hab_area"),
+                                str_hab_area,
+                            ]
+                        )
+                        .select(
+                            [
+                                "country",
+                                "str_hab_area",
+                                "eff_pot_hab_area",
+                                "connected_eff_pot_hab_area",
+                                "occupied_hab_area",
+                                "pa_area",
+                                "state",
+                            ]
+                        )
+                        .updateMask(self.scl_image.select("country").eq(country_code))
+                    )
+                    mode_band = ["country"]
+                    sum_bands = [
+                        "str_hab_area",
+                        "eff_pot_hab_area",
+                        "connected_eff_pot_hab_area",
+                        "occupied_hab_area",
+                        "pa_area",
+                    ]
+                    reducer = (
+                        ee.Reducer.mode()
+                        .forEach(mode_band)
+                        .combine(ee.Reducer.sum().forEach(sum_bands))
+                    )
+
+                    def get_stats(item):
+                        item = ee.Dictionary(item)
+                        country_id = item.get("country")
+                        state_id = item.get("state")
+                        country_name = (
+                            self.countries.filter(
+                                ee.Filter.eq("isonumeric", country_id)
+                            )
+                            .first()
+                            .get("iso2")
+                        )
+                        country_name = ee.String(
+                            ee.Algorithms.If(
+                                ee.String(country_name).equals("Jammu and Kashmir"),
+                                "IN",
+                                country_name,
+                            )
+                        )
+                        state_name = (
+                            self.states.filter(ee.Filter.eq("gadm1code", state_id))
+                            .first()
+                            .get("gadm1name")
+                        )
+                        c = country.get("iso2")
```

Comment:
Hmmm again I get a little lost trying to follow all the country property manipulations... seems like there ought to be one place where `iso2` etc. get defined.